types/ui/dialogButton.d.ts

```diff
@@ -0,0 +1,16 @@
+declare class DialogButton<T = unknown> {
```

Comment:
Foundry does not provide a class `DialogButton`, does it? So maybe this should be an `interface` instead? And Is there even a reason to move it to a separate file or shouldn't it rather be in the `Dialog` namespace and be called `Button`?

Reply:
It is there. Line 13007.

Reply:
Yeah, it's described in the JSDoc. But there is no class that is available via the API. you cannot do
```ts
const myButton = new DialogButton();
```
Declaring it as a class here would make that compile and result in a runtime error.

We are using the same pattern of declaring these documented types, that do not exist as classes, as interfaces in namespaces for other things, too, e.g. `ActiveEffect.Data` (see foundry.js line 35391)`. Actually the stuff for `ActiveEffect` and `DialogButton` are the only places where foundry.js uses the `@typedef` JSDoc.

Reply:
Yeah you are right, that should definitely be changed to a global type instead. I will put that in the `types.d.ts`.

Reply:
Are you sure it shouldn't just be in the `Dialog` namespace?

Reply:
But seeing as it is only used there and we already did it that way for `Application.HeaderButton`, I'll just move it to the `Dialog` namespace.

Reply:
It's the same with `ActiveEffectData` (= `ActiveEffect.Data` in our types), `ActiveEffectDuration` (= `ActiveEffect.Duration`) and `ActiveEffectChange` (= `ActiveEffect.Change`). If you want, I can also move all of them to be consistent with the type definitions of foundry.js.